doc/source/glossary.rst

```diff
@@ -344,122 +315,98 @@ Glossary
 
 
    Fortran order
-       See `column-major`
-
-
-   flattened
-       Collapsed to a one-dimensional array. See `numpy.ndarray.flatten`
-       for details.
+       Same as `column-major`
 
 
-   homogeneous
-       Describes a block of memory comprised of blocks, each block comprised of
-       items and of the same size, and blocks are interpreted in exactly the
-       same way. In the simplest case each block contains a single item, for
-       instance int32 or float64.
-
-
-   immutable
-       An object that cannot be modified after execution is called
-       immutable.  Two common examples are strings and tuples.
 
-
-   itemsize
-       The size of the dtype element in bytes.
+   flattened
+       See :term:`ravel`.
 
 
-   list
-       A Python container that can hold any number of objects or items.
-       The items do not have to be of the same type, and can even be
-       lists themselves::
 
-         >>> x = [2, 2.0, "two", [2, 2.0]]
+   homogeneous
+       All elements of a homogeneous array have the same type. ndarrays, in
+       contrast to Python lists, are homogeneous. The type can be complicated,
+       as in a :term:`structured array`, but all elements have that type.
 
-       The list `x` contains 4 items, each which can be accessed individually::
+      NumPy `object arrays <#term-object-array>`_, which contain references to
+      Python objects, fill the role of heterogeneous arrays.
```

Comment:
Indent is inconsistent here, and will likely render badly.

Also, should be ``:term:`object-array` ``